Api/Services/MessageService.cs

```diff
@@ -0,0 +1,200 @@
+
+namespace Reservant.Api.Services;
+
+/// <summary>
+/// Service for managing messages.
+/// </summary>
+public class MessageService(
+    UserManager<User> userManager,
+    ApiDbContext dbContext,
+    ValidationService validationService)
+{
+     /// <summary>
+    /// Updates a message by Id.
+    /// </summary>
+    /// <param name="messageId"> id of a messfe</param>
+    /// <param name="request"> update request</param>
+    /// <param name="userId">ID of the user making the request</param>
+    /// <returns></returns>
+    public async Task<Result<MessageVM>> UpdateMessageAsync(int messageId, UpdateMessageRequest request, string userId)
+    {
+        var message = await dbContext.Messages
+        .Include(m => m.Author)
+            .FirstOrDefaultAsync(m => m.Id == messageId);
+
+        if (message == null)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = null,
+                ErrorMessage = "message not found",
+                ErrorCode = ErrorCodes.NotFound
+            };
+        }
+
+        if (message.AuthorId!=userId)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = null,
+                ErrorMessage = "message was not send by provided user",
+                ErrorCode = ErrorCodes.AccessDenied
+            };
+        }
+
+        message.Contents = request.Contents;
+    
+        var validationResult = await validationService.ValidateAsync(message, userId);
+        if (!validationResult.IsValid)
+        {
+            return validationResult;
+        }
+
+        dbContext.Messages.Update(message);
+        await dbContext.SaveChangesAsync();
+
+        return new MessageVM
+        {
+            Id = message.Id,
+            Contents = message.Contents,
+            DateSent = message.DateSent,
+            DateRead = message.DateRead,
+            AuthorsFirstName = message.Author.FirstName,
+            AuthorsLastName = message.Author.LastName,
+            MessageThreadId = message.MessageThreadId
+        };
+    }
+
+
+    /// <summary>
+    /// marks message as read
+    /// </summary>
+    /// <param name="messageId">Id of the message</param>
+    /// <param name="userId">Id of the message</param>
+    /// <returns>marks message as read</returns>
+    public async Task<Result<MessageVM>> MarkMessageAsReadByIdAsync(int messageId, string userId)
+    {
+        var message = await dbContext.Messages
+        .Include(m => m.Author)
+        .Include(m => m.MessageThread)
+            .ThenInclude(m => m.Participants)
+        .FirstOrDefaultAsync(m => m.Id == messageId);
+
+        if (message == null)
+        {
+            return new ValidationFailure
+            {
+                PropertyName = null,
+                ErrorMessage = "Message not found",
+                ErrorCode = ErrorCodes.NotFound
+            };
+        }
+
+        
+        if (message.DateRead != null)
```

Comment:
Myślę, że to nie powinien być błąd, tylko jeśli DateRead już jest ustawione, to nic nie róbmy